### PR TITLE
Update MISRA config

### PR DESCRIPTION
**Pull request by AniruddhaKanhere**
*Issue #, if available:*

*Description of changes:*
This PR updates the coverity config file
Adds conditional compilation option to test CMake.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -31,64 +37,65 @@ set( CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib )
 
 #  ========================== Coverity Analysis Configuration =======================================
 
-# Coverity can run off the unit test targets, but this avoids calling unnecessary
-# unit testing tools like gcov and CMock.
+if( COV_ANALYSIS )
```
Comment (chinglee-iot) [inline test/CMakeLists.txt:40]: Can you help to describe when to use `COV_ANALYSIS` and `UNITTEST` variables?
Do we also update the [coverity README](https://github.com/aws/Device-Shadow-for-AWS-IoT-embedded-sdk/blob/main/tools/coverity/README.md) to use `COV_ANALYSIS`?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -20,6 +20,7 @@ jobs:
           -G "Unix Makefiles" \
           -DCMAKE_BUILD_TYPE=Debug \
           -DBUILD_CLONE_SUBMODULES=ON \
+          -DUNITTEST=1 \
```
Comment (chinglee-iot) [inline .github/workflows/ci.yml:23]: This question may be out of the scope of this PR.
Do we need to add `UNITTEST` variable in other library as well?

Comment (AniruddhaKanhere) [reply .github/workflows/ci.yml:23]: Which other library are you talking about?

File: test/CMakeLists.txt (hunk 1 above)
Comment (AniruddhaKanhere) [reply test/CMakeLists.txt:40]: Updated! Nice catch, thanks!

**Review by moninom1 — APPROVED** [review]

**Review by Skptak — APPROVED** [review]

**Review by paulbartell — APPROVED** [review]

**Pull request merged**